plugins/reolink/src/main.ts

```diff
@@ -91,7 +91,7 @@ class ReolinkCamera extends RtspSmartCamera implements Camera, Reboot, Intercom,
         }
 
         try {
-            const ai: AIState = this.storageSettings.values.hasObjectDetector[0]?.value;
+            const ai: AIState = this.storageSettings.values.hasObjectDetector?.value;
```

Comment:
this change looks overly permissive. need to see the camera payload.

Reply:
This is the payload coming out of the camera.
```
{
  value: {
    channel: 0,
    dog_cat: { alarm_state: 0, support: 1 },
    face: { alarm_state: 0, support: 0 },
    people: { alarm_state: 0, support: 1 },
    vehicle: { alarm_state: 0, support: 1 }
  },
  data: [ { cmd: 'GetAiState', code: 0, value: [Object] } ]
}
```

Reply:
This is a different payload, as it is not enclosed by a JSON array. Change needs to be made to check both the original doorbell payload as well as this new payload (removing the [0] index).

Reply:
I don't see a JSON array coming back for either my POE doorbell or these cameras.

Was this originally coded against an NVR?